packages/core/src/browser/context-menu-renderer.ts

```diff
@@ -16,22 +16,73 @@
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
+import { injectable } from 'inversify';
 import { MenuPath } from '../common/menu';
+import { Disposable, DisposableCollection } from '../common/disposable';
 
 export type Anchor = MouseEvent | { x: number, y: number };
 
 export function toAnchor(anchor: HTMLElement | { x: number, y: number }): Anchor {
     return anchor instanceof HTMLElement ? { x: anchor.offsetLeft, y: anchor.offsetTop } : anchor;
 }
 
-export interface ContextMenuAccess {
+export abstract class ContextMenuAccess implements Disposable {
+
+    protected readonly toDispose = new DisposableCollection();
+    readonly onDispose = this.toDispose.onDispose;
+
+    constructor(toClose: Disposable) {
+        this.toDispose.push(toClose);
+    }
+
+    get disposed(): boolean {
+        return this.toDispose.disposed;
+    }
+
+    dispose(): void {
+        this.toDispose.dispose();
+    }
+
 }
 
-export const ContextMenuRenderer = Symbol('ContextMenuRenderer');
-export interface ContextMenuRenderer {
+@injectable()
+export abstract class ContextMenuRenderer {
+
+    protected _current: ContextMenuAccess | undefined;
+    protected readonly toDisposeOnSetCurrent = new DisposableCollection();
+    /**
+     * Currently opened context menu.
+     * Rendering a new context menu will close the current.
+     */
+    get current(): ContextMenuAccess | undefined {
+        return this._current;
+    }
+    protected setCurrent(current: ContextMenuAccess | undefined): void {
+        if (this._current === current) {
+            return;
+        }
+        this.toDisposeOnSetCurrent.dispose();
+        this._current = current;
+        if (current) {
+            this.toDisposeOnSetCurrent.push(current.onDispose(() => {
+                this._current = undefined;
+            }));
+            this.toDisposeOnSetCurrent.push(current);
+        }
+    }
+
     render(options: RenderContextMenuOptions): ContextMenuAccess;
     /** @deprecated since 0.7.2 pass `RenderContextMenuOptions` instead */
     render(menuPath: MenuPath, anchor: Anchor, onHide?: () => void): ContextMenuAccess;
+    render(arg: MenuPath | RenderContextMenuOptions, arg2?: Anchor, arg3?: () => void): ContextMenuAccess {
```

Comment:
`render(menuPath: MenuPath | RenderContextMenuOptions, anchor?: Anchor, onHide?: () => void): ContextMenuAccess`

Reply:
yeah, except the first argument

Reply:
I see. You're right, please pick a better name than `arg`.

Reply:
`pathOrOptions`?

Reply:
went with `menuPathOrOptions`